research/deep_speech/deep_speech.py

```diff
@@ -0,0 +1,269 @@
+import tensorflow as tf
+# pylint: enable=g-bad-import-order
+
+import deep_speech_model
+import data.dataset as dataset
+from official.utils.export import export
+from official.utils.flags import core as flags_core
+from official.utils.logs import hooks_helper
+from official.utils.logs import logger
+from official.utils.misc import distribution_utils
+from official.utils.misc import model_helpers
+
+
+def convert_keras_to_estimator(keras_model, num_gpus, flags_obj):
+  """Configure and convert keras model to Estimator.
+  Args:
+    keras_model: A Keras model object.
+    num_gpus: An integer, the number of gpus.
+    flags_obj: An object containing parsed flags. See define_resnet_flags()
+      for details.
+  Returns:
+    estimator: The converted Estimator.
+  """
+  optimizer = tf.keras.optimizers.SGD(
+      lr=flags_obj.learning_rate, momentum=flags_obj.momentum,
+      decay=flags_obj.l2, nesterov=True)
+  keras_model.compile(
+      optimizer=optimizer, loss={"ctc_loss": lambda y_true, y_pred: y_pred})
+
+  distribution_strategy = distribution_utils.get_distribution_strategy(
+      num_gpus)
+  run_config = tf.estimator.RunConfig(
+      train_distribute=distribution_strategy, session_config=session_config)
+
+  estimator = tf.keras.estimator.model_to_estimator(
+      keras_model=keras_model, model_dir=flags_obj.model_dir, config=run_config)
+
+  return estimator
+
+
+def generate_dataset(data_dir):
+  """Generate a speech dataset."""
+  audio_conf = dataset.AudioConfig(
+      flags_obj.sample_rate, flags_obj.frame_length, flags_obj.frame_step)
+  train_data_conf = dataset.DatasetConfig(
+      audio_conf,
+      data_dir,
+      flags_obj.vocabulary_file,
+  )
+  speech_dataset = dataset.DeepSpeechDataset(train_data_conf)
+  return speech_dataset
+
+
+def main(_):
+  with logger.benchmark_context(flags_obj):
+    run_deep_speech(flags_obj)
+
+
+def run_deep_speech(_):
+  """Run deep speech training and eval loop."""
+  # Data preprocessing
+  # The file name of training and test dataset
+  tf.logging.info("Data preprocessing...")
+  train_speech_dataset = generate_dataset(flags_obj.train_data_dir)
+  eval_speech_dataset = generate_dataset(flags_obj.eval_data_dir)
+
+  # Number of label classes. Label string is "abcdefghijklmnopqrstuvwxyz' -"
+  num_classes = len(train_speech_dataset.speech_labels)
+
+  # Input shape of each data example: [time_steps (T), feature_bins(F), channel(C)]
+  # Channel is set as 1 by default.
+  input_shape = (None, train_speech_dataset.num_feature_bins, 1)
+
+  # Create deep speech model and convert it to Estimator
+  tf.logging.info("Creating Estimator from Keras model...")
+  keras_model = deep_speech_model.DeepSpeech(
+      input_shape, flags_obj.rnn_hidden_layers, flags_obj.rnn_type,
+      flags_obj.is_bidirectional, flags_obj.rnn_hidden_size,
+      flags_obj.rnn_activation, num_classes, flags_obj.use_bias)
+  # Check model structure
+  tf.logging.info(keras_model.summary(line_length=100))
```

Comment:
What does this output? Do we want to have that in all cases, or was this part of the debugging process?